simphony/cuds/primitive_cell.py

```diff
@@ -0,0 +1,537 @@
+import numpy as np
+from enum import Enum
+
+
+class BravaisLattice(Enum):
+    """The 3D Bravais lattices"""
+    CUBIC = 0
+    BODY_CENTERED_CUBIC = 1
+    FACE_CENTERED_CUBIC = 2
+    RHOMBOHEDRAL = 3
+    HEXAGONAL = 4
+    TETRAGONAL = 5
+    BODY_CENTERED_TETRAGONAL = 6
+    ORTHORHOMBIC = 7
+    BODY_CENTERED_ORTHORHOMBIC = 8
+    FACE_CENTERED_ORTHORHOMBIC = 9
+    BASE_CENTERED_ORTHORHOMBIC = 10
+    MONOCLINIC = 11
+    BASE_CENTERED_MONOCLINIC = 12
+    TRICLINIC = 13
+
+
+class PrimitiveCell(object):
+    """
+    A primitive cell of a Bravais lattice.
+    The cell edges are represented with primitive vectors.
+    
+    Primitive cells can be configured either by providing the
+    primitive vectors explicitly (default) or by providing appropriate
+    (conventional) unit cell edge parameters (for the class methods).
+
+    Attributes
+    ----------
+    p1, p2, p3: 3 x 3D vector
+        primitive vectors
+
+    bravais_lattice: BravaisLattice(Enum)
+        the 3D Bravais lattice for which the primitive cell is defined,
+        i.e. not the symmetry group or type of the primitive cell itself
+    """
+   
+    def __init__(self, p1, p2, p3):
```

Comment:
You can have an optional argument for the lattice kind that is set to `None`. This will remove the uncomfortable access of the `self._bravais_lattice` in the class methods and the arbitrary value in line 82.